node-webservice/react-client/src/Components/GameplayPage.jsx

```diff
@@ -131,6 +104,12 @@ class GameplayPage extends Component {
       const newState = messageData.gameState;
       this.setState({ gameplayPhase: newState });
     });
+
+    socket.on("game-message", messageData => {
+      if(messageData.action == "currentParts" && messageData.name == this.state.username){
+        this.setState({parts: JSON.parse(JSON.stringify(messageData.parts))});
```

Comment:
Not too clear why the parts object needs to be turned into a string and then back into an object. A comment here about why would be good